README.md

```diff
@@ -58,7 +90,94 @@ defmodule PossumLodge do
 end
 ```
 
-## Tests
+`with_lock` also allows setting an optional timeout and expiry in milliseconds.
+
+```elixir
+defmodule PossumLodge do
+  use RedisMutex
+
+  def get_oauth do
+    with_lock("my_key", 2000, 4000) do
+      "Quando omni flunkus moritati"
+    end
+  end
+end
+```
+
+
+## Testing your application with `RedisMutex`
+
+### Testing your application with Redis running
+
+If you are running Redis when you are running your test suite, simply having the `redis_mutex` config set and running the
+default command works:
+
+```
+mix test
+```
+
+### Testing your application without an instance of Redis running
+
+If you want to test your application without an instance of Redis running, you will need to define a double for
+`RedisMutex.Lock` and specify that double as the `lock_module` in the `redis_mutex` config in `config/test.exs`. This
+could be a module that implements the behaviour of `RedisMutex.Lock` or it could be a double using `Mox` or another
+library for test doubles. The examples here assume the use of `Mox`.
+
+#### Define a mock for `RedisMutex.Lock`
+
+If you are using `Mox`, you can define the mock along with your other mocks.
+
+```
+Mox.defmock(RedisMutexLockMock, for: RedisMutex.Lock)
+```
+
+#### Set the `lock_module` in your application's configuration
+
+In `config/test.exs`, set your mock as the `lock_module` for `redis_mutex`.
+
+```
+config :redis_mutex,
+  lock_module: RedisMutexLockMock
+```
+
+#### Define stubs for the lock module's `with_lock` 
+
+Depending on whether or not you set a timeout and/or expiry when using `with_lock`, define one or more stubs that
+fit the arity of `with_lock` that you are using.
+
+```elixir
+    stub(RedisMutexLockMock, :with_lock, fn _key, do_clause ->
```

Comment:
I rarely use stub when I use Mox. I much prefer using the expect and return the value. seems more of what is needed on the place that needs it

Reply:
This has been removed in favor of a more generic "you can provide a double if you want"